nautobot_device_lifecycle_mgmt/jobs/cve_tracking.py

```diff
@@ -57,3 +71,324 @@ def run(self, published_after, debug=False):  # pylint: disable=too-many-locals,
+                last_modified_date=date.fromisoformat(info.get("modified_date", "1900-01-01")[0:10]),
+                link=info["url"],
+                cvss=info["cvss_base_score"],
+                severity=info["cvss_severity"],
+                cvss_v2=info["cvssv2_score"],
+                cvss_v3=info["cvssv3_score"],
+                comments="ENTRY CREATED BY NAUTOBOT NIST JOB",
+            )
+
+            if created:
+                created_count += 1
+
+        self.logger.info("Created New CVEs.", extra={"grouping": "CVE Creation"})
+
+    def get_cve_info(self, cpe_software_search_urls: list, software_id=None) -> dict:
+        """Search NIST for software and related CVEs."""
+        for cpe_software_search_url in cpe_software_search_urls:
+            result = self.query_api(cpe_software_search_url)
+
+            all_cve_info = {"new": {}, "existing": {}}
+            if result["totalResults"] > 0:
+                self.logger.info(
+                    "Received %s results.",
+                    result["totalResults"],
+                    extra={"object": SoftwareVersion.objects.get(id=software_id), "grouping": "CVE Creation"},
+                )
+                cve_list = [cve["cve"] for cve in result["vulnerabilities"]]
+                dlc_cves = [cve.name for cve in CVELCM.objects.all()]
+
+                all_cve_info = self.process_cves(cve_list, dlc_cves, software_id)
+
+        return all_cve_info
+
+    def process_cves(self, cve_list, dlc_cves, software_id):
+        """Method to return processed CVE info.
+
+        Args:
+            cve_list (list): List of CVE returned from CPE search
+            dlc_cves (list): List of all DLM CVE objects
+            software_id (object): UUID of the Software being queried
+
+        Returns:
+            dict: Dictionary of CVEs in either new or existing categories
+        """
+        processed_cve_info = {"new": {}, "existing": {}}
+        if cve_list:
+            for cve in cve_list:
+                cve_name = cve["id"]
+                if cve_name.startswith("CVE"):
+                    if cve_name not in dlc_cves:
+                        processed_cve_info["new"].update({cve_name: self.prep_cve_for_dlc(cve)})
+                    else:
+                        processed_cve_info["existing"].update({cve_name: self.prep_cve_for_dlc(cve)})
+            self.logger.info(
+                "Prepared %s CVE for creation." % len(processed_cve_info["new"]),
+                extra={"object": SoftwareVersion.objects.get(id=software_id), "grouping": "CVE Creation"},
+            )
+
+        return processed_cve_info
+
+    def query_api(self, url):
+        """Establishes a session for use of retries and backoff.
+
+        Args:
+            url (string): The API endpoint getting queried.
+
+        Returns:
+            dict: Dictionary of returned results if successful.
+        """
+        try:
+            result = self.session.get(url, headers=self.headers)
```

Comment:
We'll set headers when we init session.

```suggestion
            result = self.session.get(url)
```